airflow/providers/amazon/aws/example_dags/example_eks_templated.py

```diff
@@ -49,50 +44,53 @@
 """
 
 with DAG(
-    dag_id='to-publish-manuals-templated',
-    default_args={'cluster_name': "{{ dag_run.conf['cluster_name'] }}"},
+    dag_id='example_eks_templated',
     schedule_interval=None,
     start_date=datetime(2021, 1, 1),
-    catchup=False,
-    max_active_runs=1,
     tags=['example', 'templated'],
+    catchup=False,
     # render_template_as_native_obj=True is what converts the Jinja to Python objects, instead of a string.
     render_template_as_native_obj=True,
 ) as dag:
-    SUBNETS = os.environ.get('EKS_DEMO_SUBNETS', 'subnet-12345ab subnet-67890cd').split(' ')
-    VPC_CONFIG = {
-        'subnetIds': SUBNETS,
-        'endpointPublicAccess': True,
-        'endpointPrivateAccess': False,
-    }
+
+    CLUSTER_NAME = "{{ dag_run.conf['cluster_name'] }}"
+    NODEGROUP_NAME = "{{ dag_run.conf['nodegroup_name'] }}"
+    VPC_CONFIG = json.loads("{{ dag_run.conf['resources_vpc_config'] }}")
```

Comment:
@ferruzzi you get the error because this line doesn't do what you expect.

You expect it `"{{ dag_run.conf['resources_vpc_config'] }}"` to be the Json define in the beginning of the file:
```
{
        "subnetIds": ["subnet-12345ab", "subnet-67890cd"],
        "endpointPublicAccess": true,
        "endpointPrivateAccess": false
}
```
    
But this is not what is happening.  The Jinja engine runs only in templated fields of operators. It does not run in top level python code. So when you execute `json.loads` you actually execute it on the string that you wrote.
If you will run in your python console :
`json.loads("{{ dag_run.conf['resources_vpc_config'] }}")`
You will see it will give you :
`json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)`

Reply:
Thanks.  Yeah, that is what I was gathering but I haven't come across an answer yet.  Jinja is new to me so this may be really obvious.  Does/can jinja magically know it should be a dict, or do I change the operator to accept type Union[dict, str], and if it's a string then handle the conversion inside the operator?

Reply:
To render as native Python object you will need `render_template_as_native_obj=True`
docs https://airflow.apache.org/docs/apache-airflow/stable/concepts/operators.html#rendering-fields-as-native-python-objects.

The templating is done as part of creating a task instance:
https://github.com/apache/airflow/blob/fded2ca0b9c995737b401896b89e5c9fd7f24c91/airflow/models/taskinstance.py#L1419

Reply:
Should be fixed in https://github.com/apache/airflow/pull/21523/commits/bc1bca40db832675533ec2a65d2a28dca9f5cffe